modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -383,10 +383,7 @@ BaseBounds getSpanBounds() {
 
     private GlyphList[] getRuns() {
         if (textRuns != null) return textRuns;
-        if (isSpan()) {
-            /* List of run is initialized when the TextFlow layout the children */
-            getParent().layout();
-        } else {
+        if (!isSpan()) {
```

Comment:
Removing the call to `getParent().layout()` here seems very likely to cause a regression in behavior. This will need some further analysis and lots of testing.

Reply:
I agree the change is quite dangerous. 
But the current behavior is in my opinion "broken by design". 
updating bounds should never trigger layout. Group also has similar behavior, which causes issues from time to time.

Reply:
I have the feeling, that this change changes much less behavior than expected.